app/controllers/hiring_staff/vacancies/copy_controller.rb

```diff
@@ -0,0 +1,25 @@
+class HiringStaff::Vacancies::CopyController < HiringStaff::Vacancies::ApplicationController
+  def create
+    vacancy = Vacancy.find(vacancy_id)
+    vacancy_copy = vacancy.dup
+    update_fields(vacancy_copy)
+    if vacancy_copy.save
+      redirect_to review_path(vacancy_copy)
+    else
+      redirect_to school_path, notice: I18n.t('errors.jobs.unable_to_copy')
+    end
+  end
+
+  private
+
+  def update_fields(vacancy)
```

Comment:
Wrapping this logic up and moving it to a small service object would get it out of the controller which would keep this file simpler and give us a way to add light unit tests for it. 

The existing condition of `if vacancy.publish_on <= Time.zone.today` is currently tested with 2 dedicated feature specs. As feature tests are expensive we should try to push as many as we can into unit tests whilst keeping our confidence high that we'd catch a regression, I think these are perfect candidates:

```
scenario 'the publish_on date is updated to today'
scenario 'the publish_on date is the same as the pending job it was copied from'
```